iotedgedev/decorators.py

```diff
@@ -0,0 +1,75 @@
+import hashlib
+import sys
+from functools import wraps
+
+
+def with_telemetry(func):
+    @wraps(func)
+    def _wrapper(*args, **kwargs):
+        from . import telemetry
+        from .telemetryconfig import TelemetryConfig
+
+        config = TelemetryConfig()
+        config.check_firsttime()
+        params = parse_params(*args, **kwargs)
+        telemetry.start(func.__name__, params)
+        try:
+            value = func(*args, **kwargs)
+            telemetry.success()
+            telemetry.flush()
```

Comment:
Here and line 25, Since telemetry.flush() happens no matter what, why not put it once in a finally block?